models/recently_pushed.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package models
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"code.gitea.io/gitea/models/activities"
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/models/user"
+
+	"xorm.io/builder"
+)
+
+type RecentlyPushedBranches struct {
+	Repo     *repo.Repository
+	BaseRepo *repo.Repository
+	RefName  string
+	Time     time.Time
+}
+
+// GetRecentlyPushedBranches returns all actions where a user recently pushed but no PRs are created yet.
+func GetRecentlyPushedBranches(ctx context.Context, u *user.User) (recentlyPushedBranches []*RecentlyPushedBranches, err error) {
+	limit := time.Now().Add(-24 * time.Hour).Unix()
+
+	actions := []*activities.Action{}
+	// We're fetching the last three commits activity actions within the limit...
+	err = db.GetEngine(ctx).
+		Select("action.ref_name, action.repo_id, action.created_unix").
+		Join("LEFT", "pull_request", "pull_request.head_branch = replace(action.ref_name, 'refs/heads/', '')").
+		Join("LEFT", "issue", "pull_request.issue_id = issue.id").
+		Join("LEFT", "repository", "action.repo_id = repository.id").
+		Where(builder.And(
+			builder.Eq{"action.op_type": activities.ActionCommitRepo},
+			// ...done by the current user
+			builder.Eq{"action.act_user_id": u.ID},
+			// ...which have been pushed to a fork or a branch different from the default branch
+			builder.Or(
+				builder.Expr("repository.default_branch != replace(action.ref_name, 'refs/heads/', '')"),
+				builder.Eq{"repository.is_fork": true},
+			),
+			// ...and don't have an open or closed PR corresponding to that branch.
+			builder.Or(
+				builder.IsNull{"pull_request.id"},
+				builder.And(
+					builder.Eq{"pull_request.has_merged": false},
+					builder.Eq{"issue.is_closed": true},
+					builder.Gt{"action.created_unix": "issue.closed_unix"},
```

Comment:
You have to use `builder.Expr("action.created_unix = issue.closed_unix")`.

Reply:
Done